src/test/java/com/solvd/laba/UserRegistrationTest.java

```diff
@@ -0,0 +1,77 @@
+package com.solvd.laba;
+
+import java.util.Map;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import com.solvd.laba.pages.*;
+import com.solvd.laba.utils.UserFactory;
+
+public class UserRegistrationTest extends BaseTest {
+
+    @Test
+    public void testUserRegistration() {
+        Map<String, String> userData = UserFactory.createUserData();
+
+        openHomePage();
+
+        logger.info("➡️ Step 2: Navigating Login Page");
+        LoginPage loginPage = homePage.header().clickSignupLogin();
+        Assert.assertTrue(loginPage.isOpened(), "Login page should be opened");
+
+        logger.info("📝 Step 3: Entering name and email for registration");
+        loginPage.signUp(userData.get("name"), userData.get("email"));
+
+        SignupPage signupPage = new SignupPage(driver);
+        Assert.assertTrue(signupPage.isOpened(), "Signup page should be opened");
+
+        logger.info("🧾 Step 4: Filling in account information");
+        signupPage.fillAccountInfo(
+                userData.get("password"),
+                userData.get("birth_day"),
+                userData.get("birth_month"),
+                userData.get("birth_year")
+        );
+
+        logger.info("🏠 Step 5: Filling in address and contact information");
+        signupPage.fillAddressInfo(
+                userData.get("name"),
+                userData.get("last_name"),
+                userData.get("company"),
+                userData.get("address1"),
+                userData.get("address2"),
+                userData.get("country"),
+                userData.get("state"),
+                userData.get("city"),
+                userData.get("zipcode"),
+                userData.get("mobile")
+        );
+
+        signupPage.clickCreateAccount();
```

Comment:
For consistency, method names could be more descriptive. For example, methods like clickCreateAccount could be named submitRegistrationForm, or fillAccountInfo could be fillRegistrationDetails